File: test/CodeCracker.Test/Verifiers/CodeFixTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.CodeAnalysis.CodeFixes;
using Microsoft.CodeAnalysis.Diagnostics;

namespace TestHelper
{
    public abstract class CodeFixTest<T, U> : CodeFixVerifier
        where T : DiagnosticAnalyzer, new()
        where U : CodeFixProvider, new()
    {
        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new T();
        }

        protected override CodeFixProvider GetCSharpCodeFixProvider()
        {
            return new U();
        }        
    }
}

File: test/VisualBasic/CodeCracker.Test/Verifiers/CodeFixTest.vb
Imports Microsoft.CodeAnalysis.CodeFixes
Imports Microsoft.CodeAnalysis.Diagnostics

Namespace TestHelper
    Public MustInherit Class CodeFixTest(Of T As {DiagnosticAnalyzer, New}, U As {CodeFixProvider, New})
        Inherits CodeFixVerifier

        Protected Overrides Function GetDiagnosticAnalyzer() As DiagnosticAnalyzer
            Return New T()
        End Function

        Protected Overrides Function GetBasicCodeFixProvider() As CodeFixProvider
            Return New U()
        End Function
    End Class
End Namespace